examples/brute_force/brute_force.cpp

```diff
@@ -25,74 +27,54 @@ struct Dummy
-  static KOKKOS_FUNCTION Point get(Dummy const &, size_type i)
+  static KOKKOS_FUNCTION size_type size(Dummy<DIM> const &d) { return d.count; }
+  static KOKKOS_FUNCTION auto get(Dummy<DIM> const &, size_type i)
   {
-    return {{(float)i, (float)i, (float)i}};
+    ArborX::ExperimentalHyperGeometry::Point<DIM> point;
+    for (int d = 0; d < DIM; ++d)
+      point[d] = (float)i;
+    return point;
   }
 };
 
-template <>
-struct ArborX::AccessTraits<Dummy, ArborX::PredicatesTag>
+template <int DIM>
+struct ArborX::AccessTraits<Dummy<DIM>, ArborX::PredicatesTag>
 {
   using memory_space = MemorySpace;
   using size_type = typename MemorySpace::size_type;
-  static KOKKOS_FUNCTION size_type size(Dummy const &d) { return d.count; }
-  static KOKKOS_FUNCTION auto get(Dummy const &, size_type i)
+  static KOKKOS_FUNCTION size_type size(Dummy<DIM> const &d) { return d.count; }
+  static KOKKOS_FUNCTION auto get(Dummy<DIM> const &, size_type i)
   {
-    return attach(
-        intersects(Sphere{{{(float)i, (float)i, (float)i}}, (float)i}), i);
+    ArborX::ExperimentalHyperGeometry::Point<DIM> center;
+    for (int d = 0; d < DIM; ++d)
+      center[d] = (float)i;
+    return attach(intersects(ArborX::ExperimentalHyperGeometry::Sphere<DIM>{
+                      center, (float)i}),
+                  i);
   }
 };
 
-int main(int argc, char *argv[])
+template <int DIM>
+void run(int nprimitives, int nqueries, int nrepeats)
 {
-  Kokkos::ScopeGuard guard(argc, argv);
-
-  int nqueries;
-  int nprimitives;
-  int nrepeats;
-  namespace bpo = boost::program_options;
-  bpo::options_description desc("Allowed options");
-  // clang-format off
-  desc.add_options()
-      ( "help", "help message" )
-      ( "predicates", bpo::value<int>(&nqueries)->default_value(5), "number of predicates" )
-      ( "primitives", bpo::value<int>(&nprimitives)->default_value(5), "number of primitives" )
-      ( "iterations", bpo::value<int>(&nrepeats)->default_value(1), "number of iterations" )
-      ;
-  // clang-format on
-  bpo::variables_map vm;
-  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
-  bpo::notify(vm);
-
-  if (vm.count("help") > 0)
-  {
-    std::cout << desc << '\n';
-    return 1;
-  }
-  printf("Primitives: %d\n", nprimitives);
-  printf("Predicates: %d\n", nqueries);
-  printf("Iterations: %d\n", nrepeats);
-
-  ARBORX_ASSERT(nprimitives > 0);
-  ARBORX_ASSERT(nqueries > 0);
-
   ExecutionSpace space{};
-  Dummy primitives{nprimitives};
-  Dummy predicates{nqueries};
+  Dummy<DIM> primitives{nprimitives};
+  Dummy<DIM> predicates{nqueries};
+
+  using Box = ArborX::ExperimentalHyperGeometry::Box<DIM>;
 
   for (int i = 0; i < nrepeats; i++)
   {
     unsigned int out_count;
     {
       Kokkos::Timer timer;
-      ArborX::BoundingVolumeHierarchy<MemorySpace> bvh{space, primitives};
+      ArborX::BasicBoundingVolumeHierarchy<MemorySpace, Box> bvh{space,
```

Comment:
Don't like this, but `BVH` and `BoundingVolumeHierarchy` don't have the `BoundingVolume` template parameter. That would have to be cleaned, probably at the same time as APIv2.